static/js/main.js

```diff
@@ -612,49 +606,199 @@ var Overlay = L.Class.extend({
         
         this._overlayElement.addEventListener('touchend', function(e) {
             var snapThreshold = -self._overlayHeight * 0.3; // Snap when dragged beyond 30% of the overlay height
+
+            // Under threshold -> snap back to the bottom
             if (parseInt(self._overlayElement.style.bottom) < snapThreshold) {
-                self._overlayElement.style.bottom = '-90%'; // Snap back to 20% visible
-                self._overlayElement.style.borderRadius = '15px 15px 0px 0px';
-                self._overlayElement.style.overflowY = 'hidden';
-                self._overlayElement.scrollTo(0, 0);
-                self._line.style.display = 'block';
-                // self._nearYouTopP.style.opacity = 0;
-            } else {
-                self._overlayElement.style.bottom = '0';
-                self._overlayElement.style.borderRadius = '0px 0px 0px 0px';
-                self._overlayElement.style.overflowY = 'auto';
-                self._line.style.display = 'none';
-                // self._nearYouTopP.style.opacity = 1;
+                self.closeOverlay();
+            }
+            // Above threshold -> reveal the overlay 
+            else {
+                self.openOverlay();
             }
         });
     },
   
-    // these are the default options
+    // Default options
     options: {
         isActive: false,
     },
     
     createStickerDiv: function (i) {
```

Comment:
Most of your code is quite tedious. Perhaps we should switch to some other framework which eases this type of code? Let's just finish this PR with plain python, but this does not scale that well. Perhaps you have some ideas to smoothen this later.

Reply:
Now that I think of it, plain python with html/css is quite simple. A solution to the comment I made above is probably just dividing files up. Thinking out loud here!